src/checkers/inference/model/ConstraintManager.java

```diff
@@ -129,6 +130,15 @@ public ExistentialConstraint createExistentialConstraint(Slot slot,
                 ifNotExistsConstraints, getCurrentLocation());
     }
 
+    /**
+     * Create an {@link ArithmeticConstraint} for the given operation and slots.
+     */
+    public ArithmeticConstraint createArithmeticConstraint(ArithmeticOperationKind operation,
+            Slot leftOperand, Slot rightOperand, Slot result) {
```

Comment:
Could `result` be other types other than `ArithmeticSlot`? If not, better to change the type from `Slot` to `ArithmeticSlot` to achieve better type safety.

Same question for `addArithemticConstraint()`.

Same question for `CombineConstraint` case. @topnessman

Reply:
True for me. Will change here and also in `ArithmeticConstraint` and `ConstraintEncoderCoordinator`.